sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-enqueuer.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,237 @@
+import { Job, Queue } from "bull";
+import { createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, addDays } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import { AssessmentWorkflowEnqueuerScheduler } from "../assessment-workflow-enqueuer.scheduler";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeStudentAssessment,
+  createFakeUser,
+  getQueueProviderName,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { Not } from "typeorm";
+import {
+  Application,
+  AssessmentTriggerType,
+  StudentAssessmentStatus,
+  User,
+} from "@sims/sims-db";
+import { StartAssessmentQueueInDTO } from "@sims/services/queue";
+
+describe(
+  describeProcessorRootTest(QueueNames.AssessmentWorkflowEnqueuer),
+  () => {
+    let app: INestApplication;
+    let db: E2EDataSources;
+    let processor: AssessmentWorkflowEnqueuerScheduler;
+    let startApplicationAssessmentQueueMock: Queue<StartAssessmentQueueInDTO>;
+    let auditUser: User;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource } = await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      auditUser = await db.user.save(createFakeUser());
+      // Mocked StartApplicationAssessment queue.
+      startApplicationAssessmentQueueMock = app.get(
+        getQueueProviderName(QueueNames.StartApplicationAssessment),
+      );
+      // Processor under test.
+      processor = app.get(AssessmentWorkflowEnqueuerScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.resetAllMocks();
+      // Ensure that all assessments will be cancelled.
+      await db.studentAssessment.update(
+        {
+          studentAssessmentStatus: Not(StudentAssessmentStatus.Cancelled),
+        },
+        { studentAssessmentStatus: StudentAssessmentStatus.Cancelled },
+      );
+    });
+
+    it(`Should queue an assessment when an active application has only the original assessment and it is in '${StudentAssessmentStatus.Submitted}' state.`, async () => {
+      // Arrange
+
+      // Application submitted with original assessment.
+      const application = await createDefaultApplication();
```

Comment:
Trying to understand the purpose of the new method? won't it be able to achieve by `saveFakeApplicationDisbursements` ?

Reply:
It is just a centralization inside the same TC file. There are positive and negative scenarios using the same method which makes it better to assume that, when there is a negative test it would be more accurate because the positive is already using the same method.

Reply:
These test cases are also targeting student assessments, not disbursements. Can you please point out the benefit of using the `saveFakeApplicationDisbursements`?
@andrepestana-aot FYI